pingcap/tidb
# cmd/importer: support probability in building incremental column

### What problem does this PR solve? 

Generate column values in incremental order mostly, but with some exceptions controlled by input parameter.

### What is changed and how it works?

Support new rule `probability=xxx` which should be used together with `incremental`, to control the exceptions in overall incremental order.

### Check List 

Tests 

 - Manual test (add detailed scripts or steps below)

```
run script:
./bin/importer -t "create table t(a date comment '[[incremental=1;repeats=3;step=-1;probability=80]]');" -c 1 -n 10 -P 4000

check table rows:
mysql> select * from t;
+------------+
| a          |
+------------+
| 2019-05-14 |
| 2019-05-02 |
| 2019-05-13 |
| 2019-06-05 |
| 2019-05-13 |
| 2019-05-12 |
| 2019-05-12 |
| 2019-08-26 |
| 2019-05-11 |
| 2019-05-11 |
+------------+
10 rows in set (0.00 sec)
```

Code changes

N/A

Side effects

N/A

Related changes

N/A
